### PR TITLE
feat(sdk): adding domain support to the ReactApp

### PR DESCRIPTION
**Pull request by marciocadev**
* Refactoring the domain
* Adding domain support to the ReactApp

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Review by tsuf239 — COMMENTED** [review]
Can you please add wing tests?

File: libs/wingsdk/src/target-tf-aws/react-app.ts [hunk 1]
```diff
@@ -15,7 +16,7 @@ import * as ex from "../ex";
  */
 export class ReactApp extends ex.ReactApp {
   private _host: cloud.Website;
-  constructor(scope: Construct, id: string, props: ex.ReactAppProps) {
+  constructor(scope: Construct, id: string, props: aws.AwsReactAppProps) {
```
Comment (tsuf239) [inline libs/wingsdk/src/target-tf-aws/react-app.ts:19]: following #4543, can you please change it to `AwsReactAppOptions`?  (also the other props if possible 🙏 )

**Review by tsuf239 — COMMENTED** [review]
Looks great! 
is it possible to add a test file written in wing?

File: libs/wingsdk/src/target-tf-aws/react-app.ts (hunk 1 above)
Comment (marciocadev) [reply libs/wingsdk/src/target-tf-aws/react-app.ts:19]: @tsuf239 

I follow this [PR](https://github.com/winglang/wing/pull/4740) and this [issue](https://github.com/winglang/wing/issues/4543)

**Review by tsuf239 — APPROVED** [review]

**Pull request merged**